### PR TITLE
Add WindowsProfile field

### PR DESCRIPTION
**Pull request by cmurphy**
Adds support for `WindowsProfile` which is required for Windows node pools.

https://github.com/rancher/rancher/issues/32586

### CODE REVIEW

**Review by mjura — DISMISSED** [review]
LGTM

**Review by mjura — DISMISSED** [review]
LGTM

File: controller/aks-cluster-config-handler.go [hunk 1]
```diff
@@ -390,6 +390,12 @@ func (h *Handler) validateConfig(config *aksv1.AKSClusterConfig) error {
 		if np.OsType == "" {
 			return fmt.Errorf(cannotBeNilError, "NodePool.OsType", config.ClusterName)
 		}
+		// Validate name length here because the AKS API does not validate it but will mysteriously fail if not in compliance.
+		// https://docs.microsoft.com/en-us/azure/aks/windows-faq#what-can-i-name-my-windows-node-pools
+		maxLengthPoolName := 6
```
Comment (thedadams) [inline controller/aks-cluster-config-handler.go:395]: Make this a constant

**Review by thedadams — CHANGES_REQUESTED** [review]

File: controller/aks-cluster-config-handler.go (hunk 1 above)
Comment (cmurphy) [reply controller/aks-cluster-config-handler.go]: done

**Review by thedadams — APPROVED** [review]

File: controller/aks-cluster-config-handler.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ const (
 	aksConfigActivePhase     = "active"
 	aksConfigUpdatingPhase   = "updating"
 	aksConfigImportingPhase  = "importing"
+	maxLengthPoolName        = 6
```
Comment (cbron) [inline controller/aks-cluster-config-handler.go:34]: nit: 
```suggestion
	poolNameMaxLength        = 6
```

Comment (cbron) [reply controller/aks-cluster-config-handler.go:34]: I think we should enforce this in the UI also, if you can make that happen

**Review by cbron — APPROVED** [review]
lgtm, with a request

File: controller/aks-cluster-config-handler.go (hunk 2 above)
Comment (cmurphy) [reply controller/aks-cluster-config-handler.go]: changed the constant name

**Review by thedadams — APPROVED** [review]

**Review by mjura — APPROVED** [review]
LGTM

File: controller/aks-cluster-config-handler.go [hunk 3]
```diff
@@ -390,6 +391,11 @@ func (h *Handler) validateConfig(config *aksv1.AKSClusterConfig) error {
 		if np.OsType == "" {
 			return fmt.Errorf(cannotBeNilError, "NodePool.OsType", config.ClusterName)
 		}
+		// Validate name length here because the AKS API does not validate it but will mysteriously fail if not in compliance.
+		// https://docs.microsoft.com/en-us/azure/aks/windows-faq#what-can-i-name-my-windows-node-pools
+		if np.OsType == "Windows" && len(*np.Name) > poolNameMaxLength {
```
Comment (cbron) [inline controller/aks-cluster-config-handler.go:396]: nit: "Windows" could be a constant

**Review by cbron — APPROVED** [review]

File: controller/aks-cluster-config-handler.go (hunk 2 above)
Comment (cmurphy) [reply controller/aks-cluster-config-handler.go]: Added to UI issue https://github.com/rancher/rancher/issues/31315#issuecomment-839970611

**Pull request merged**